### PR TITLE
Dịch "sequence_vn" - Phần 4

### LINKED ISSUE #1459: Dịch "sequence_vn" - Phần 4
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_recurrent-neural-networks/sequence_vn.md#L200-L323

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by davidnvq**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #1459 

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [ ] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

3. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_recurrent-neural-networks/sequence_vn.md [hunk 1]
```diff
@@ -269,7 +280,15 @@ A few layers of a fully connected network, ReLU activation and $\ell_2$ loss.
 Since much of the modeling is identical to the previous sections when we built regression estimators in Gluon, we will not delve into much detail.
 -->
 
-*dịch đoạn phía trên*
+Tiếp theo, chúng ta cần biến chuỗi thời gian này thành các đặc trưng và nhãn được sử dụng để huấn luyện mạng.
+Dựa trên kích thước embedding $\tau$, chúng ta ánh xạ dữ liệu thành các cặp $y_t = x_t$ và $\mathbf{z}_t = (x_{t-1}, \ldots, x_{t-\tau})$.
+Sẽ rất tinh tế nếu bạn đọc có thể nhận thấy rằng hiện tại chúng ta mới có $\tau$ điểm dữ liệu ít hơn hơn, vì chúng ta không có đủ lịch sử về $\tau$ đầu tiên trong số đó.
+Chúng ta loại bỏ một vài phần tử trên để khắc phục điều này, đặc biệt nếu chuỗi thời gian rất dài.
```
Comment (cuongvng) [inline chapter_recurrent-neural-networks/sequence_vn.md:286]:
```suggestion
Một cách đơn giản để khắc phục điều này là loại bỏ vài phần tử đó, đặc biệt nếu chuỗi thời gian rất dài.
```

File: chapter_recurrent-neural-networks/sequence_vn.md [hunk 2]
```diff
@@ -269,7 +280,15 @@ A few layers of a fully connected network, ReLU activation and $\ell_2$ loss.
...
+Hoặc là, chúng ta có thể đệm 0 vào chuỗi thời gian.
```
Comment (cuongvng) [inline chapter_recurrent-neural-networks/sequence_vn.md:287]:
```suggestion
Hoặc cũng có thể đệm 0 vào chuỗi thời gian.
```

File: chapter_recurrent-neural-networks/sequence_vn.md [hunk 3]
```diff
@@ -269,7 +280,15 @@ A few layers of a fully connected network, ReLU activation and $\ell_2$ loss.
...
+Mã nguồn dưới đây về cơ bản là giống hệt với mã nguồn huấn luyện trong các phần trước.
+Chúng tôi cố gắng giữ cho kiến trúc đơn giản.
+Một vài tầng của mạng kết nối đầy đủ, hàm kích hoạt ReLU và hàm mất mát $ \ ell_2 $.
```
Comment (cuongvng) [inline chapter_recurrent-neural-networks/sequence_vn.md:290]:
```suggestion
Một vài tầng của mạng kết nối đầy đủ, hàm kích hoạt ReLU và hàm mất mát $\ell_2$.
```

File: chapter_recurrent-neural-networks/sequence_vn.md [hunk 4]
```diff
@@ -301,7 +320,7 @@ loss = gluon.loss.L2Loss()
 Now we are ready to train.
 -->
 
-*dịch đoạn phía trên*
+Bây giờ thì chúng ta đã sẵn sàng để huấn luyện.
```
Comment (cuongvng) [inline chapter_recurrent-neural-networks/sequence_vn.md:323]:
```suggestion
Bây giờ chúng ta đã sẵn sàng huấn luyện.
```

File: chapter_recurrent-neural-networks/sequence_vn.md [hunk 5]
```diff
@@ -499,7 +518,7 @@ với dấu `@` ở đầu. Ví dụ: @aivivn.
 *
 
 <!-- Phần 4 -->
-*
+* Nguyễn Văn Quang
```
Comment (duythanhvn) [inline chapter_recurrent-neural-networks/sequence_vn.md:521]:
```suggestion
* Nguyễn Văn Quang
* Nguyễn Văn Cường
```

File: chapter_recurrent-neural-networks/sequence_vn.md [hunk 6]
```diff
@@ -227,21 +228,31 @@ For more on this topic see e.g., the book by :cite:`Peters.Janzing.Scholkopf.201
 We are barely scratching the surface of it.
 -->
 
-*dịch đoạn phía trên*
+Trên thực tế, nếu chúng ta có một mô hình Markov, chúng ta cũng có được phân phối xác suất có điều kiện ngược.
+Tuy nhiên, trong nhiều trường hợp, có tồn tại một hướng nhất định của dữ liệu, cụ thể là luôn tiến về phía trước theo thời gian.
```
Comment (lkhphuc) [inline chapter_recurrent-neural-networks/sequence_vn.md:232]:
```suggestion
Tuy nhiên trong nhiều trường hợp, tự nhiên dữ liệu có tồn tại một trật tự nhất định, cụ thể là luôn tiến về phía trước theo thời gian.
```

File: chapter_recurrent-neural-networks/sequence_vn.md [hunk 7]
```diff
@@ -227,21 +228,31 @@ For more on this topic see e.g., the book by :cite:`Peters.Janzing.Scholkopf.201
...
+Rõ ràng là các sự kiện trong tương lai không thể ảnh hưởng đến quá khứ.
+Do đó, nếu chúng ta thay đổi $x_t$, thì có thể ảnh hưởng đến những gì xảy ra với $x_{t+1}$ trong tương lai, nhưng không ảnh hưởng tới quá khứ ở chiều ngược lại.
+Nếu chúng ta thay đổi $x_t$, phân phối trên các sự kiện trong quá khứ sẽ không thay đổi.
+Do đó, việc giải thích $p(x_{t+1} \mid x_t)$ sẽ đơn giản hơn là $p(x_t \mid x_{t+1})$.
+Ví dụ: bài báo :cite:`Hoyer.Janzing.Mooij.ea.2009` chỉ ra rằng trong một số trường hợp chúng ta có thể tìm $x_{t+1} = f(x_t) + \epsilon$ khi có thêm nhiễu, trong khi đó, điều ngược lại không đúng. Đây là một tin tuyệt vời vì chúng ta thường quan tâm tới hướng truyền xuôi để ước lượng.
```
Comment (lkhphuc) [inline chapter_recurrent-neural-networks/sequence_vn.md:237]:
```suggestion
Ví dụ: bài báo :cite:`Hoyer.Janzing.Mooij.ea.2009` chỉ ra rằng trong một số trường hợp chúng ta có thể tìm $x_{t+1} = f(x_t) + \epsilon$ khi có thêm nhiễu, trong khi đó, điều ngược lại không đúng. 
Đây là một tin tuyệt vời vì chúng ta thường quan tâm tới việc ước lượng theo hướng đi tới hơn.
```

File: chapter_recurrent-neural-networks/sequence_vn.md [hunk 8]
```diff
@@ -269,7 +280,15 @@ A few layers of a fully connected network, ReLU activation and $\ell_2$ loss.
 Since much of the modeling is identical to the previous sections when we built regression estimators in Gluon, we will not delve into much detail.
 -->
 
-*dịch đoạn phía trên*
+Tiếp theo, chúng ta cần biến chuỗi thời gian này thành các đặc trưng và nhãn được sử dụng để huấn luyện mạng.
+Dựa trên kích thước embedding $\tau$, chúng ta ánh xạ dữ liệu thành các cặp $y_t = x_t$ và $\mathbf{z}_t = (x_{t-1}, \ldots, x_{t-\tau})$.
+Để ý kĩ, có thể thấy rằng chúng ta sẽ mất $\tau$ điểm dữ liệu đầu tiên, vì không có đủ $\tau$ dữ liệu trong quá khứ của chúng.
```
Comment (lkhphuc) [inline chapter_recurrent-neural-networks/sequence_vn.md:285]:
```suggestion
Để ý kĩ, có thể thấy rằng chúng ta sẽ mất $\tau$ điểm dữ liệu đầu tiên, vì chúng không có đủ $\tau$ điểm dữ liệu trong quá khứ.
```

File: chapter_recurrent-neural-networks/sequence_vn.md [hunk 9]
```diff
@@ -269,7 +280,15 @@ A few layers of a fully connected network, ReLU activation and $\ell_2$ loss.
...
+Một cách đơn giản để khắc phục điều này là loại bỏ vài phần tử đó, đặc biệt nếu chuỗi thời gian rất dài.
+Hoặc cũng có thể đệm 0 vào chuỗi thời gian.
```
Comment (lkhphuc) [inline chapter_recurrent-neural-networks/sequence_vn.md:287]:
```suggestion
Một cách khác là đệm giá trị 0 vào chuỗi thời gian.
```

File: chapter_recurrent-neural-networks/sequence_vn.md [hunk 10]
```diff
@@ -269,7 +280,15 @@ A few layers of a fully connected network, ReLU activation and $\ell_2$ loss.
...
+Mã nguồn dưới đây về cơ bản là giống hệt với mã nguồn huấn luyện trong các phần trước.
+Chúng tôi cố gắng giữ cho kiến trúc đơn giản.
+Một vài tầng của mạng kết nối đầy đủ, hàm kích hoạt ReLU và hàm mất mát $\ell_2$.
```
Comment (lkhphuc) [inline chapter_recurrent-neural-networks/sequence_vn.md:290]:
```suggestion
Chúng tôi cố gắng giữ cho kiến trúc đơn giản với vài tầng kết nối đầy đủ, hàm kích hoạt ReLU và hàm mất mát $\ell_2$.
```

File: chapter_recurrent-neural-networks/sequence_vn.md [hunk 11]
```diff
@@ -269,7 +280,15 @@ A few layers of a fully connected network, ReLU activation and $\ell_2$ loss.
...
+Do phần lớn mô hình giống hệt với các bộ ước lượng hồi quy viết bằng Gluon trong các phần trước, nên chúng ta sẽ không đi sâu vào chi tiết.
```
Comment (lkhphuc) [inline chapter_recurrent-neural-networks/sequence_vn.md:291]:
```suggestion
Do việc mô hình hoá phần lớn là giống với khi ta xây dựng các bộ ước lượng hồi quy viết bằng Gluon trong các phần trước, nên chúng ta sẽ không đi sâu vào chi tiết trong phần này.
```

File: chapter_recurrent-neural-networks/sequence_vn.md [hunk 12]
```diff
@@ -499,7 +518,8 @@ với dấu `@` ở đầu. Ví dụ: @aivivn.
 *
 
 <!-- Phần 4 -->
-*
+* Nguyễn Văn Quang
+* Nguyễn Văn Cường
```
Comment (duythanhvn) [inline chapter_recurrent-neural-networks/sequence_vn.md:522]:
```suggestion
* Nguyễn Văn Cường
* Lê Khắc Hồng Phúc
```

File: chapter_recurrent-neural-networks/sequence_vn.md [hunk 13]
```diff
@@ -227,21 +228,32 @@ For more on this topic see e.g., the book by :cite:`Peters.Janzing.Scholkopf.201
...
+Trên thực tế, nếu có một mô hình Markov, chúng ta cũng có thể thu được một phân phối xác suất có điều kiện ngược.
+Tuy nhiên trong nhiều trường hợp, tự nhiên dữ liệu có tồn tại một trật tự nhất định, cụ thể là luôn tiến về phía trước theo thời gian.
+Rõ ràng là các sự kiện trong tương lai không thể ảnh hưởng đến quá khứ.
+Do đó, nếu chúng ta thay đổi $x_t$, thì có thể ảnh hưởng đến những gì xảy ra với $x_{t+1}$ trong tương lai, nhưng không ảnh hưởng tới quá khứ ở chiều ngược lại.
+Nếu chúng ta thay đổi $x_t$, phân phối trên các sự kiện trong quá khứ sẽ không thay đổi.
+Do đó, việc giải thích $p(x_{t+1} \mid x_t)$ sẽ đơn giản hơn là $p(x_t \mid x_{t+1})$.
+Ví dụ: bài báo :cite:`Hoyer.Janzing.Mooij.ea.2009` chỉ ra rằng trong một số trường hợp chúng ta có thể tìm $x_{t+1} = f(x_t) + \epsilon$ khi có thêm nhiễu, trong khi đó, điều ngược lại không đúng. 
+Đây là một tin tuyệt vời vì chúng ta thường quan tâm tới việc ước lượng theo hướng đi tới hơn.
+Để tìm hiểu thêm về chủ đề này, có thể tìm đọc cuốn sách :cite:`Peters.Janzing.Scholkopf.2017`.
+Chúng ta sẽ chỉ tìm hiểu sơ qua trong phần này.
```
Comment (rootonchair) [inline chapter_recurrent-neural-networks/sequence_vn.md:240]:
```suggestion
Chúng ta nãy giờ chỉ mới nói về bề mặt của vấn đề này.
```

**Comment by davidnvq** [comment]
Định mấy hôm nữa vào sửa một thể mà nay hầu hết PR mn đã giải quyết xong xuôi rồi. Cảm ơn cả nhà đã review cực kỳ cẩn thận PR này và tất cả các PR khác ạ. Cảm ơn sếp @duythanhvn  đã thay mặt resolve từng suggestion commit.  Mình xin comment một cái này tránh bão noti mn :)))
